Models/articles-model.js

```diff
@@ -16,17 +16,28 @@ function fetchArticleByID(article_id){
     })
 }
 
-function fetchArticles(){
-    let query = `
-    SELECT articles.author, articles.title, articles.article_id, articles.topic, articles.created_at, articles.votes, articles.article_img_url, SUM(comments.article_id) AS comment_count 
+function fetchArticles(topic){
+
+    const queryValues = [];
+
+    let query = `SELECT articles.author, articles.title, articles.article_id, articles.topic, articles.created_at, articles.votes, articles.article_img_url, COUNT(comments.article_id)::int AS comment_count 
     FROM articles 
     LEFT JOIN comments 
-    ON articles.article_id = comments.article_id 
-    GROUP BY articles.author, articles.title, articles.article_id, articles.topic, articles.created_at, articles.votes, articles.article_img_url 
-    ORDER BY created_at DESC;
-    `;
+    ON articles.article_id = comments.article_id `;
+
+    if (topic) {
+        queryValues.push(topic);
+        query += `WHERE articles.topic = $1 `;
+    } 
 
-    return db.query(query).then(({rows}) => {
+    query += `GROUP BY articles.article_id `;
+
+    query += `ORDER BY created_at DESC;`;
+
+    return db.query(query, queryValues).then(({rows}) => {
+        if(rows[0] === undefined && topic){
+            return Promise.reject({status : 404, msg : "topic does not exist"})
```

Comment:
potentially here you could get nothing back from this query when the topic exists but there's no articles for it yet, it would be better to have another check at this point before you Promise.reject to see if it exists or not